### ISSUE TITLE
Provide a perl6-like hyper op in examples/threads

### ISSUE THREAD

**Issue by rurban** [parrot/parrot#889]
In order to change perl6 to support threads, rakudo wants to see more threads examples, 
something like rakudos hyper implemented in pir.

```
my @y = @m »*« @x »+« @b;
```

**Comment by FROGGS** [comment]
**FROGGS:** in fact I still don't know how to do a non blocking lexical lookup
**FROGGS:** (I am just talking about looking up variables, and subroutine to do some computation, I'm not even talking about writing to a lexical)
**rurban:** check if the type is a Proxy, follow the proxy then, if not read it directly. if you need to write to a proxy, schedule a writer task

**Comment by FROGGS** [comment]
**FROGGS:** rurban: IIRC the lexical lookup *was* blocking... you say this is not the case anymore? lexical lookup will happily run in any worker thread?
**rurban:** FROGGS: every data access is lock-free, just not wait-free

**Comment by rurban** [comment]
Yes, only the nqp variant of the lexpad pmc is broken, the parrot variant works ok.
`src/vm/parrot/pmc/nqplexpad.pmc` vs `src/pmc/lexpad.pmc`
